app/Http/Controllers/Maps/DeviceDependencyController.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * DependencyController.php
+ *
+ * Controller for graphing Relationships
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @copyright  2019 Thomas Berberich
+ * @author     Thomas Berberich <sourcehhdoctor@gmail.com>
+ */
+
+namespace App\Http\Controllers\Maps;
+
+use App\Models\Device;
+use Illuminate\Http\Request;
+use LibreNMS\Util\Url;
+
+class DeviceDependencyController extends MapController
+{
+    protected static function deviceList($request)
+    {
+        $group_id = $request->get('group');
+
+        if (! $group_id) {
+            return Device::hasAccess($request->user())->with('parents', 'location')->get();
+        }
+
+        $devices = Device::inDeviceGroup($group_id)
+            ->hasAccess($request->user())
+            ->with([
+                'location',
+                'parents' => function ($query) use ($request) {
+                    $query->hasAccess($request->user());
+                },
+                'children' => function ($query) use ($request) {
+                    $query->hasAccess($request->user());
+                }])
+            ->get();
+
+        return $devices->merge($devices->map->only('children', 'parents')->flatten());
```

Comment:
add `->loadMissing('parents', 'location');` to this

Reply:
@Jellyfrog 
Don't know how you want me to implement this.
Everywhere and with every combination of `parent`, `location`, `childrens` some Devices or Relationgraphs are missing.

Reply:
Sorry it needs to be added after so it loads the data missing after the merge

Reply:
do you mean to change it to:
`return $devices->merge($devices->map->only('children', 'parents')->loadMissing('parents', 'location')->flatten());`
This doesn't work at me. It throws following Error:

> production.ERROR: Method Illuminate\Support\Collection::loadMissing does not exist. {"userId":2,"exception":"[object] (BadMethodCallException(code: 0): Method Illuminate\\Support\\Collection::loadMissing does not exist. at /opt/librenms/vendor/laravel/framework/src/Illuminate/Support/Traits/Macroable.php:104)
[stacktrace]
#0 /opt/librenms/app/Http/Controllers/Maps/DeviceDependencyController.php(54): Illuminate\\Support\\Collection->__call('loadMissing', Array)
#1 /opt/librenms/app/Http/Controllers/Maps/DeviceDependencyController.php(68): App\\Http\\Controllers\\Maps\\DeviceDependencyController::deviceList(Object(Illuminate\\Http\\Request))
#2 [internal function]: App\\Http\\Controllers\\Maps\\DeviceDependencyController->dependencyMap(Object(Illuminate\\Http\\Request))
#3 /opt/librenms/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(54): call_user_func_array(Array, Array)

Reply:
@Jellyfrog 
i changed it now to:
`return $devices->merge($devices->map->only('children', 'parents')->flatten())->loadMissing('parents', 'location');`
Hope it's right now :-)